### PR TITLE
[video_player] Make sure the plugin is correctly initialized

### PR DESCRIPTION
**Pull request by cbenhagen**
## Description

This seemed to have no effect. Was it optimized away because `_` was not used?
```dart
final VideoPlayerPlatform _ = VideoPlayerPlatform.instance..init();
```

## Related Issues

Fixes https://github.com/flutter/flutter/issues/47524

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by cbenhagen** [comment]
@cyanglaz can I ask you for a review and a tip on how to properly test this?

**Comment by cbenhagen** [comment]
Added a test to check if init has been called.

**Comment by cbenhagen** [comment]
@mklim do you have an idea why the test works locally but fails on Cirrus CI? Is there maybe a better way to test this?

**Comment by cbenhagen** [comment]
@amirh people keep running into issues with the released video_player plugin which already have PRs with simple fixes pending (this and https://github.com/flutter/plugins/pull/2397). Can you help me get the plugin into a good state again or do you know someone else I could ping for a quick review?

**Comment by amirh** [comment]
> This seemed to have no effect. Was it optimized away because _ was not used?

Yes, static initializers are lazily evaluated when the field is first used.

File: packages/video_player/video_player/lib/video_player.dart [hunk 1]
```diff
@@ -14,10 +14,10 @@ import 'package:video_player_platform_interface/video_player_platform_interface.
...
     show DurationRange, DataSourceType, VideoFormat;
 
-// This will clear all open videos on the platform when a full restart is
-// performed.
-// ignore: unused_element
-final VideoPlayerPlatform _ = VideoPlayerPlatform.instance..init();
+final VideoPlayerPlatform _videoPlayerPlatform = VideoPlayerPlatform.instance
+  // This will clear all open videos on the platform when a full restart is
+  // performed.
+  ..init();
```
Comment (amirh) [inline packages/video_player/video_player/lib/video_player.dart]: I don't like this hack in general, but we don't have a better alternative right now. I filed https://github.com/flutter/flutter/issues/48586 to expose an engine restart hook to the plugin's platform code.

File: packages/video_player/video_player/test/video_player_test.dart [hunk 2]
```diff
@@ -91,6 +91,14 @@ void main() {
       fakeVideoPlayerPlatform = FakeVideoPlayerPlatform();
     });
 
+    test('plugin initialized', () async {
```
Comment (amirh) [inline packages/video_player/video_player/test/video_player_test.dart]: This test seems to fail on CI.
I think a previous test case may already be triggering the initialization and thus this one doesn't.

I'm not sure if package:test allows controlling the order of test execution. Maybe we can split this test case to it's own file to ensure it's the first (and only) test for the given engine instance?

Comment (cbenhagen) [reply packages/video_player/video_player/test/video_player_test.dart]: Sounds good. I have no idea how I would do this unfortunately.

Comment (amirh) [reply packages/video_player/video_player/test/video_player_test.dart]: Create another file named `video_player_initialization_test.dart` with just this test

**Comment by cbenhagen** [comment]
Tests pass. Changelog and version already adapted in the hope that https://github.com/flutter/plugins/pull/2397 would land first.

File: packages/video_player/video_player/test/video_player_initialization_test.dart [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+// found in the LICENSE file.
+
+import 'package:flutter/widgets.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:video_player/video_player.dart';
+
+import 'video_player_test.dart' show FakeVideoPlayerPlatform;
+
+void main() {
+  test('plugin initialized', () async {
```
Comment (amirh) [inline packages/video_player/video_player/test/video_player_initialization_test.dart]: nit: add a comment explaining that this test should be the only test in this file as it needs to run first

Comment (cbenhagen) [reply packages/video_player/video_player/test/video_player_initialization_test.dart]: Done

**Comment by cbenhagen** [comment]
@amirh does this need to be reviewed by another person?

**Comment by amirh** [comment]
I was just waiting for the tree to become green before landing (was red for multiple days until now 😞  )

**Comment by cbenhagen** [comment]
Oh ok. Thanks for merging!

**Pull request merged**